src/sentry/relay/config/__init__.py

```diff
@@ -213,47 +195,113 @@ def _get_project_config(project, full_config=True, project_keys=None):
                         )
                     active_rules.append(rule)
 
-            cfg["config"]["dynamicSampling"] = {"rules": active_rules}
+            return {"rules": active_rules}
+
+    return None
+
+
+def add_experimental_config(
+    config: MutableMapping[str, Any],
+    key: str,
+    function: Callable[..., Any],
+    *args: Any,
+    **kwargs: Any,
+) -> None:
+    """Try to set cfg[key] = function(*args, **kwargs).
```

Comment:
nit:
```suggestion
    """Try to set `config[key] = function(*args, **kwargs)`.
```
`cfg` -> `config` because of the arg name.